src/stdlib_stats_distribution_uniform.fypp

```diff
@@ -0,0 +1,526 @@
+#:include "common.fypp"
+#:set RC_KINDS_TYPES = REAL_KINDS_TYPES + CMPLX_KINDS_TYPES
+#:set ALL_KINDS_TYPES = INT_KINDS_TYPES + RC_KINDS_TYPES
+module stdlib_stats_distribution_uniform
+    use stdlib_kinds
+    use stdlib_error, only : error_stop
+    use stdlib_stats_distribution_PRNG, only : dist_rand
+
+    implicit none
+    private
+
+    real(dp), parameter  :: MESENNE_NUMBER = 1.0_dp / (2.0_dp ** 53 - 1.0_dp)
+    integer(int64), parameter :: INT_ONE = 1_int64
+
+    public :: rvs_uniform
+    public :: pdf_uniform
+    public :: cdf_uniform
+    public :: shuffle
+
+
+    interface rvs_uniform
+    !! Version experimental
+    !!
+    !! Get uniformly distributed random variate for integer, real and complex
+    !! variables.
+    !! ([Specification](../page/specs/stdlib_stats_distribution_uniform.html#
+    !! description))
+
+        module procedure unif_dist_rvs_0_rsp                 ! 0 dummy variable
+
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure unif_dist_rvs_1_${t1[0]}$${k1}$     ! 1 dummy variable
+        #:endfor
+
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure unif_dist_rvs_${t1[0]}$${k1}$       ! 2 dummy variables
+        #:endfor
+
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure unif_dist_rvs_array_${t1[0]}$${k1}$ ! 3 dummy variables
+        #:endfor
+    end interface rvs_uniform
+
+
+    interface pdf_uniform
+    !! Version experimental
+    !!
+    !! Get uniform distribution probability density (pdf) for integer, real and
+    !! complex variables.
+    !! ([Specification](../page/specs/stdlib_stats_distribution_uniform.html#
+    !! description))
+
+        #:for k1, t1 in ALL_KINDS_TYPES
+        module procedure unif_dist_pdf_${t1[0]}$${k1}$
```

Comment:
Suggest you replace `unif_dist_pdf_` with `pdf_uniform_` where it occurs in function names, for ease of reading.